dpnp/backend/include/dpnp_iface_fptr.hpp

```diff
@@ -127,6 +127,8 @@ enum class DPNPFuncName : size_t
     DPNP_FN_MINIMUM,                  /**< Used in numpy.minimum() implementation  */
     DPNP_FN_MODF,                     /**< Used in numpy.modf() implementation  */
     DPNP_FN_MULTIPLY,                 /**< Used in numpy.multiply() implementation  */
+    DPNP_FN_MULTIPLY_ARRAY_ARRAY,     /**< Used in numpy.multiply(array, array) implementation  */
+    DPNP_FN_MULTIPLY_ARRAY_SCALAR,    /**< Used in numpy.multiply(array, scalar) implementation  */
```

Comment:
The proposal here might be - leave this code is it was before. No addition needed. Everything will be the same from "user" perspective.
One thing should be added - `size_in2`. The parameter to get information about size of each array (size will be size_in1 and one more extra parameter `size_in2`)
Down the call stack, it might be implemented as dispatcher - each combination goes to it's special internal backend function with specific kernel.
From user perspective - minor change and no extra public functions.
Internally - more kernels for each combination (array+scalar, array+array, etc)
Internal implementation might be changed in future without public interface changes.

Reply:
This is one of possible "acceptable" implementation. This is not quite beauty but it will work. Also, many other libraries implemented in this way (one public function and many internal specializations).
The main limitation factor I see here - we can't specialize it in compile time. Templates usage might be better solution but we have no information in compile time about parameters (array or scalar).

Reply:
So, I see you are going in right direction. Some cleanup of this PR needed

Reply:
We can't remove `DPNP_FN_MULTIPLY_ARRAY_SCALAR` because we have different return type for input `array*array` and `array*scalar` of the same type.

Reply:
Let the dispatcher function handle this, and remove this from public API.